plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+#include "BaseStateDriver.cpp"
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Joint.hh>
+#include <gz/sim/Link.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/Sensor.hh>
+#include <gz/sim/System.hh>
+#include <gz/sim/Util.hh>
+#include <gz/sim/components/Sensor.hh>
+#include <gz/transport/Node.hh>
+#include <iostream>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/PolyDriverList.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+namespace gzyarp
+{
+
+class BaseState : public System, public ISystemConfigure, public ISystemPostUpdate
+{
+public:
+    BaseState()
+        : m_deviceRegistered(false)
+    {
+    }
+
+    virtual ~BaseState()
+    {
+        if (m_deviceRegistered)
+        {
+            Handler::getHandler()->removeDevice(m_deviceScopedName);
+            m_deviceRegistered = false;
+        }
+
+        if (m_baseStateDriver.isValid())
+            m_baseStateDriver.close();
+        BaseStateDataSingleton::getBaseStateDataHandler()->removeBaseLink(m_baseLinkScopedName);
+        yarp::os::Network::fini();
+    }
+
+    virtual void Configure(const Entity& _entity,
+                           const std::shared_ptr<const sdf::Element>& _sdf,
+                           EntityComponentManager& _ecm,
+                           EventManager& /*_eventMgr*/) override
+    {
+        yarp::os::Network::init();
+        if (!yarp::os::Network::checkNetwork())
+        {
+            yError() << "Yarp network does not seem to be available, is the yarpserver running?";
+            return;
+        }
+
+        std::string netWrapper = "analogServer";
+        ::yarp::dev::Drivers::factory().add(
+            new ::yarp::dev::DriverCreatorOf<::yarp::dev::gzyarp::BaseStateDriver>("gazebo_"
+                                                                                   "basestate",
+                                                                                   netWrapper
+                                                                                       .c_str(),
+                                                                                   "BaseStateDrive"
+                                                                                   "r"));
```

Comment:
This is quite hard to read, and it is not obvious that the two strings are actually concatenated by the compiler. Do you have any idea of how to make this more readable?

Reply:
Note that you can locally disable clang-format with https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code .

Reply:
Solved in abb966b648750631dd1496e5e4cfb4bc831d1aa5 , I defined an alias for the long type in order to have this line more readable.